### ISSUE TITLE
Arrays possibly being transposed when converting NWB files from HDF5 to ZARR

### ISSUE THREAD

**Issue by oruebel** [hdmf-dev/hdmf-zarr#45]
The tutorial for converting NWB data from HDF5 to Zarr, currently shows the following warnings (see also See https://hdmf-zarr.readthedocs.io/en/latest/tutorials/plot_convert_nwb.html#read-the-zarr-file-back-in)

<img width="952" alt="Screen Shot 2022-12-13 at 12 31 26 AM" src="https://user-images.githubusercontent.com/10999845/207265575-17469f77-908f-46ba-adcd-68275d947f6e.png">

In particular the warning ```warn("Length of data does not match length of timestamps. Your data may be transposed. Time should be on "``` is something that should be looked at, as it appears that (some) arrays may for some reason be transposed in the conversion.

**Comment by oruebel** [comment]
Using the latest version of PyNWB and HDMF locally, the warning we see currently with ```warn("Length of data does not match length of timestamps. Your data may be transposed. Time should be on "``` does not occur, but instead the following warnings are now being raised: 


```
/Users/oruebel/Devel/nwb/pynwb/src/pynwb/core.py:47: UserWarning: OpticalSeries 'StimulusPresentation_encoding': The number of frame indices in 'starting_frame' should have the same length as 'external_file'.
  warn(error_msg)
/Users/oruebel/Devel/nwb/pynwb/src/pynwb/core.py:47: UserWarning: OpticalSeries 'StimulusPresentation_encoding': Either external_file or data must be specified (not None), but not both.
  warn(error_msg)
```

Reading the data directly from HDF5 yields the same warnings, i.e., these warnings are actually due to errors in the file that is being used in the tutorial rather than indicating and error in the library.

**Comment by oruebel** [comment]
I also tested in the regular tutorial, and it looks like arrays are  indeed not being transposed. It appears that the warnings currently may be due to versions of PyNWB/HDMF used, that have already been addressed on the dev branches of those libraries. The new warnings for ``OpticalSeries`` are correct and are due to errors in the file that is being used. 

Closing this issue as it does not appear to be an issue in HDMF_ZARR.
